tests/rados/test_v1client.py

```diff
@@ -0,0 +1,103 @@
+"""
+Module to verify client connection over v1 port does not result in crashes
+"""
+
+import datetime
+import random
+import re
+import time
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    # CEPH-83594645
+    Covers:
+        - BZ-2237038
+    Test to verify client connection over v1 port does not result in crashes
+    Steps
+    1. Deploy a ceph cluster
+    2. Choose a node at random which is not already configured as a client
+    3. Copy client keyring and ceph conf file having only v1 ports
+    4. Install ceph-common package
+    5. Run ceph command using new client and monitor any crash
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    log.info(
+        "Running test case to verify client connection over v1 port does not result in crashes"
+    )
+
+    try:
+        # if rhbuild.startswith("5"):
+        #     log.info("Test is not valid for Pacific, BZ yet to be fixed/raised")
+        #     return 0
+
+        # choose an osd host at random
+        osd_hosts = rados_obj.ceph_cluster.get_nodes(role="osd")
+        osd_host = random.choice(osd_hosts)
+
+        rados_obj.configure_host_as_client(host_node=osd_host)
```

Comment:
I do not feel comfortable converting a OSD host into a client with only v1 addresses present.

OSDs and most other daemons refer ceph.conf file upon restart, for connecting to mon config db.

Though this would improve test coverage, having a separate client node would be best.

We can add a new client, and use that to perform tests here. Can be taken up as part of new task.

Reply:
An OSD node out of the box does not have ceph.conf or ceph.client.keyrings present in /etc/ceph/ so why would it matter if a ceph.conf is added with only v1 addresses, and if addition of a ceph.conf with v1 addresses causes problems with restart, we will have a bug.